### PR TITLE
backport some upstream patches

### PR DESCRIPTION
**Pull request by DioEgizio**
nothing much to say, some upstream patches backported

### CODE REVIEW

**Review by Scrumplex — APPROVED** [review]

**Review by txtsd — APPROVED** [review]

File: launcher/JavaCommon.cpp [hunk 1]
```diff
@@ -17,6 +17,17 @@ bool JavaCommon::checkJVMArgs(QString jvmargs, QWidget *parent)
             QMessageBox::Warning)->exec();
         return false;
     }
+    // block lunacy with passing required version to the JVM
+    if (jvmargs.contains(QRegExp("-version:.*"))) {
+        auto warnStr = QObject::tr(
+            "You tried to pass required java version argument to the JVM (using \"-version=xxx\"). This is not safe and will not be allowed.\n"
```
Comment (kthchew) [inline launcher/JavaCommon.cpp]:
```suggestion
            "You tried to pass required java version argument to the JVM (using \"-version:xxx\"). This is not safe and will not be allowed.\n"
```

File: launcher/JavaCommon.cpp [hunk 2]
```diff
@@ -17,6 +17,17 @@ bool JavaCommon::checkJVMArgs(QString jvmargs, QWidget *parent)
...
+            "This message will be displayed until you remove this from the JVM arguments.");
```
Comment (kthchew) [inline launcher/JavaCommon.cpp:24]: If a user tries to use the `-version:` flag, they're probably trying to select an appropriate Java version. I think the message would be more helpful if it suggested manually changing the Java version in the instance settings to accomplish this instead.

**Review by kthchew — CHANGES_REQUESTED** [review]

File: launcher/JavaCommon.cpp (hunk 2 above)
Comment (DioEgizio) [reply launcher/JavaCommon.cpp:24]: I mean yeah I just backported that patch from upstream lol

File: launcher/JavaCommon.cpp [hunk 3]
```diff
@@ -17,6 +17,17 @@ bool JavaCommon::checkJVMArgs(QString jvmargs, QWidget *parent)
             QMessageBox::Warning)->exec();
         return false;
     }
+    // block lunacy with passing required version to the JVM
+    if (jvmargs.contains(QRegExp("-version:.*"))) {
+        auto warnStr = QObject::tr(
+            "You tried to pass required java version argument to the JVM (using \"-version:xxx\"). This is not safe and will not be allowed.\n"
```
Comment (Scrumplex) [inline launcher/JavaCommon.cpp]:
```suggestion
            "You tried to pass required Java version argument to the JVM (using \"-version:xxx\"). This is not safe and will not be allowed.\n"
```

File: launcher/JavaCommon.cpp (hunk 2 above)
Comment (Scrumplex) [reply launcher/JavaCommon.cpp:24]: I think it's fine. I have read that some people use this argument to "improve" their Java options or whatever. But it seems like it causes issues, which is why upstream blocks it.

**Review by Scrumplex — APPROVED** [review]

File: launcher/Application.cpp [hunk 4]
```diff
@@ -696,6 +696,9 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
         // Minecraft launch method
         m_settings->registerSetting("MCLaunchMethod", "LauncherPart");
 
+        // Minecraft offline player name
+        m_settings->registerSetting("LastOfflinePlayerName", "");
```
Comment (d-513) [inline launcher/Application.cpp:700]: do we need this? We have separate offline accounts option, so it seems redundant. Maybe Launch Offline should be removed entirely
For now it's fine though

**Review by d-513 — APPROVED** [review]

File: launcher/Application.cpp (hunk 4 above)
Comment (DioEgizio) [reply launcher/Application.cpp:700]: Yeah for now it's fine, but for some users it's better to just click "play offline" and having an offline session

**Pull request merged**